### DISCUSSION TITLE
Risks of just learning everything by default?

### DISCUSSION THREAD

**Discussion by eaglgenes101** [dim-an/cod#52] [Q&A] [answered]
After using this extension for some time, I'm getting used to giving cod the okay to learn a command's completion. So far I haven't found an instance where I wanted cod to not learn the completion for a command. Are there risks to just setting cod to always just learn the completion for a command, no questions asked, anytime it can?

**Comment by dim-an** [comment, marked as answer]
Each time you allow `cod` to learn command, `cod` reruns it. 

For example:
 1. you just run `./purge-hard-drive --force --help`
 2. `cod` detects that command looks like help invocation and suggests to learn it
 3. You allow `cod` to learn this command
 4. `cod` runs `./purge-hard-drive --force --help` again, but this time it tries to capture command output and parse it.

I don't like idea that some utility silently echoes my commands and I want such rerunning to be explicit so in my setup I always explicitly allow `cod` to relaunch command.

But maybe this is my paranoia :) It's also very uncommon to me to disallow `cod` to learn something.

Other slight risk is `cod` learns command that already has better completion provided by other means. If this happens `cod` completion will overwrite other completion. But you always can remove unnecessary `cod` completions with `cod rm`, so it shouldn't stop you.
